src/psyclone/dynamo0p3.py

```diff
@@ -8831,6 +8837,132 @@ def stencil(self, value):
         '''
         self._stencil = value
 
+    def infer_datatype(self, symbol_table, proxy=False):
```

Comment:
The gocean implementation uses `self._call` to obtain the symbol table. Could you explain why you pass `symbol_table` as an argument instead. Is it to do with builtins not existing?

Reply:
Hmm. It must have been something to do with the order in which we construct things which I subsequently had to fix (for something else) by moving things around. In turn, that means that we can use `self._call` now so I've removed this argument.